app/src/main/java/com/capstone/weatherapp/CityListViewModel.kt

```diff
@@ -24,7 +24,7 @@ class CityListViewModel(private val weatherRepo: WeatherRepository) : ViewModel(
                 weatherRepo.refreshCityList()
                 _eventNetworkError.value = false
             } catch (e: Exception) {
-                if (cityList.value.isNullOrEmpty()) {
+                if (cityList.value == null) {
                     _eventNetworkError.value = true
```

Comment:
you shouldn't need any of this eventNetworkError code anymore, the error state should be handled by your sealed class. feel free to delete this